src/snmalloc/proxy/new.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#include "snmalloc/ds_core/defines.h"
+
+namespace snmalloc
+{
+
+  // This is used in both vendored and non-vendored mode.
+  struct PlacementToken
+  {};
+
+  inline constexpr PlacementToken placement_token{};
```

Comment:
Why is this needed?  I guess it would be nice to just define
```C++
SNMALLOC_FAST_PATH_INLINE void*
operator new(size_t, void* ptr) noexcept
{
  return ptr;
}
```
But I assume that fails?

Reply:
Placement new operator has to appear at global namespace. Consider using `snmalloc` as a header library with other `c++` programs, if we provide such definitions, then it is likely to conflict with `<new>` or other's definitions.